File: lightsabers/if_else.rb
class Jedi < ActiveRecord::Base
  has_many :lightsabers, as: :wieldable
end

class Sith < ActiveRecord::Base
  has_many :lightsabers, as: :wieldable
end

class Lightsaber < ActiveRecord::Base
  belongs_to :wieldable, polymorphic: true, after_add: :set_color

  def set_color(warrior)
    if warrior.class == Jedi
      self.color = ['green', 'blue', 'purple'].sample
    else
      self.color = 'red'
    end
  end
end

File: csharp/lightsabers/if_else.cs
"case.cs"

class Jedi
{
  ICollection _lightsabers = new Collection();

  ICollection public lightsabers
  {
    get { return _lightsabers }
    set {
     _lightsabers.Add(lightsaber);
     lightsaber.AfterAdd();
     return _lightsabers;
    }
  }
}

class Sith {
}

class Lightsaber
{
  string[] JediColors = { "green", "blue", "purple" };

  public void AfterAdd (object klass)
  {
    if (klass is Jedi) {
      this.color = JediColors[random.Next(0, JediColors.length)];
    } else {
      this.color = "red";
    }
  }
}
